client/containers/Pub/PubDocument/PubWordCountButton.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useMemo, useState } from 'react';
+import { Popover, Button, Icon } from '@blueprintjs/core';
+
+import { Node } from 'prosemirror-model';
+
+require('./pubWordCountButton.scss');
+
+type Props = {
+	doc: Node;
+};
+
+const getWordAndCharacterCountsFromDoc = (node: Node) => {
+	const text = node.textBetween(0, node.content.size, ' ', ' ');
+	const words = text.split(' ').filter((word) => word !== '');
+	return [words.length, text.length];
+};
+
+const PubHeaderFormattingWordCountButton = (props: Props) => {
+	const { doc } = props;
+	const [open, setOpen] = useState(false);
+	const counts = useMemo(() => {
+		if (open) {
+			return getWordAndCharacterCountsFromDoc(doc);
+		}
+		return null;
+	}, [open, doc]);
+	const content = (
```

Comment:
fwiw you could also write `const content = counts && <div> ... </div>` since the `null` fallback is fine for `Popover` content. A little slippery.